### PR TITLE
Generate unique token and redirect to /list

### LINKED ISSUE #3: 3. As a user, I want to set up a new shopping list so I can start tracking purchased items.
A shopping list consists of a set of items associated with a user’s token. Creating a new list consists of the following sequence:

* Generate a new, unique token
* Save the token to `localStorage`
* Show the user the list view

The following script should be used to generate a suitable token: `src/lib/tokens.js`

AC:

* For users who do not already have a token/list, a button or link exists on the home screen that allows them to create a new list
* Clicking the button/link generates a new token and saves it to `localStorage`
* Once the token has been created and saved, the user is redirected to the "list" view
* For users who do have a token already saved in `localStorage`, they should be shown/redirected to the "list" view

Notes:

* Eventually the welcome/home page will look like the following wireframe: https://github.com/the-collab-lab/smart-shopping-list/blob/main/_resources/wireframes/welcome-screen.png
* It's good for your whole team to look at the wireframe now, but be sure not to try to implement all of the visible functionality at once. There will be future stories to add the other features such as joining an existing list.
* For this story, it's sufficient just to have a simple "create list" button. Although you are not required to save the token to Firestore, some teams have opted to also save the token to Firestore when generating a token.

### PR DESCRIPTION
**Pull request by AirZT11**
## Description
* In a handleClick function, we generated a unique token using getToken from token.js and stored it in localStorage
* Used useState to set loggedIn state to true
* Based on loggedIn state, redirect to List component
* If token exists in localStorage, automatically redirect to List component when app renders

## Related Issue
Closes #3

## Acceptance Criteria
[Acceptance Criteria](https://github.com/the-collab-lab/tcl-27-smart-shopping-list/issues/3)

* For users who do not already have a token/list, a button or link exists on the home screen that allows them to create a new list
* Clicking the button/link generates a new token and saves it to localStorage
* Once the token has been created and saved, the user is redirected to the "list" view
* For users who do have a token already saved in localStorage, they should be shown/redirected to the "list" view

## Type of Changes

|     | Type                       |
| --- | -------------------------- |
|    | :bug: Bug fix              |
|  ✓ | :sparkles: New feature     |
|    | :hammer: Refactoring       |
|    | :100: Add tests            |
|    | :link: Update dependencies |
|    | :scroll: Docs              |

## Testing Steps / QA Criteria
From your terminal, pull down this branch with git pull origin sk-tc-new-shopping-list and check that branch out with git checkout sk-tc-new-shopping-list.  Then npm install or yarn to install the newly added dependency locally and npm start or yarn start to launch the app.  

Click on the "Create List..." button to generate a new token, which is stored in localStorage. Make sure to check that the app redirects to list view and that the url route changes to '/list'. Also, without clearing localStorage, refresh the app from the root url and make sure that it automatically renders list view.

### CODE REVIEW

File: src/App.js [hunk 1]
```diff
@@ -1,23 +1,45 @@
...
+    setLoggedIn(true);
+  };
+
   return (
     <div className="App">
       <GroceryContainer />
       <BottomNav />
       <Switch>
+        <Route exact path="/">
+          {loggedIn ? <Redirect to="/list" /> : null}
```
Comment (Dani-Kirby) [inline src/App.js]: Are there any other behaviors we will want to set in this ternary operator? I feel like this covers everything, but want to make sure we aren't missing anything with this expected behavior.

File: src/App.js [hunk 2]
```diff
@@ -1,23 +1,45 @@
...
+        </Route>
+
         <Route path="/list" component={List}></Route>
         <Route path="/add-an-item" component={AddItem}></Route>
       </Switch>
+
+      <button onClick={handleClick}>Create List...</button>
```
Comment (Dani-Kirby) [inline src/App.js]: Location of this button with the links means I can't click it when there are more than 9 or so items in the list. Had to delete some from the DB to confirm this behavior. We'll need to make sure we address this going forward. Should be accomplished by user story 4 when we remove the list from the main page.

**Review by Dani-Kirby — APPROVED** [review]
This looks good! We'll want to make sure that we remove the Create List button within the list view and only have it rendering on the / path. It makes a new token each time it's clicked since it's accessible from the list.

**Review by YolandaHaynes — APPROVED** [review]
It's working on my end when I tested it out! Yay! 
  
  The only thing I realized is was what @DPoplin brought up, I can still see the create list button in the list page (it confused me a bit because I didn't realized I was already redirected to the list page and initially thought the button didn't work). 
  
  Also when I'm in the local storage, every time I clicked on the create list button I get a new token. My other question is where does the hard coded token ("josef heron mudd") comes into play? 
  
Other than that, good job you guys!

File: src/App.js (hunk 1 above)
Comment (AirZT11) [reply src/App.js]: Yeah we were thinking that we could render a welcome page but we haven't built out that component just yet. For the time being, it should just render the App component if a token doesn't exist in localStorage.

File: src/App.js (hunk 2 above)
Comment (AirZT11) [reply src/App.js]: We also noticed this behavior. Should be a relatively easy fix in the css file. I'll take a stab at it and get that looking good!

File: src/App.js [hunk 3]
```diff
@@ -1,20 +1,41 @@
...
+import getToken from './lib/tokens';
+
 import GroceryContainer from './components/containers/GroceryContainer';
 import List from './components/containers/List';
 import AddItem from './components/AddItem';
 import BottomNav from './components/BottomNav';
 
 function App() {
+  const [loggedIn, setLoggedIn] = useState(false);
+
```
Comment (aprilleone) [inline src/App.js:15]: With these being consts, does that mean they can be reset? will they be able to log out?

File: src/App.js (hunk 1 above)
Comment (aprilleone) [reply src/App.js]: Another way to do this is with the && option:
https://reactjs.org/docs/conditional-rendering.html#inline-if-with-logical--operator

**Review by aprilleone — APPROVED** [review]
Fantastic! Well done team!

**Pull request merged**